src/Renderer/HorizonRenderer.cpp

```diff
@@ -278,19 +281,37 @@ HorizonRenderer::Draw(Canvas &canvas, const PixelRect &rc,
     BulkPixelPoint zero_mark[3] = {p1, p2, p3};
     canvas.DrawPolygon(zero_mark, 3);
 
-    for (int k=-20; k<=20; k+=10){
-      m1 = {-radius/5, int(h/2.f*(pitch_degrees+k)/MAX_HEIGHT_DEGREES)};
-      m2 = {+radius/5, int(h/2.f*(pitch_degrees+k)/MAX_HEIGHT_DEGREES)};
+    // pitch marks
+    int pitch_10 = (int(pitch_degrees) / 10) * 10;
+    for (int k=pitch_10-20; k<=pitch_10+20; k+=10){
+      if (k==0) continue;
+      m1 = {-radius/5, int(h/2.f*(pitch_degrees-k)/MAX_HEIGHT_DEGREES)};
+      m2 = {+radius/5, int(h/2.f*(pitch_degrees-k)/MAX_HEIGHT_DEGREES)};
       rotate(m1, center, Angle::Degrees(- bank_degrees), p1);
       rotate(m2, center, Angle::Degrees(- bank_degrees), p2);
       canvas.DrawLine(p1,p2);
+
+
+      canvas.Select(look.mark_font);
+      canvas.SetBackgroundTransparent();
+      canvas.SetTextColor( COLOR_WHITE);
+      char buffer[4];
+      sprintf(buffer, "%+3d", k);
+      canvas.DrawText(p2, buffer);
     }
-    for (int k=-25; k<=25; k+=10){
-      m1 = {-radius/10, int(h/2.f*(pitch_degrees+k)/MAX_HEIGHT_DEGREES)};
-      m2 = {+radius/10, int(h/2.f*(pitch_degrees+k)/MAX_HEIGHT_DEGREES)};
+    for (int k=pitch_10-25; k<=pitch_10+25; k+=10){
+      m1 = {-radius/10, int(h/2.f*(pitch_degrees-k)/MAX_HEIGHT_DEGREES)};
+      m2 = {+radius/10, int(h/2.f*(pitch_degrees-k)/MAX_HEIGHT_DEGREES)};
       rotate(m1, center, Angle::Degrees(- bank_degrees), p1);
       rotate(m2, center, Angle::Degrees(- bank_degrees), p2);
       canvas.DrawLine(p1,p2);
+
+      canvas.Select(look.mark_font);
+      canvas.SetBackgroundTransparent();
+      canvas.SetTextColor( COLOR_WHITE);
+      char buffer[4];
+      sprintf(buffer, "%+3d", k);
+      canvas.DrawText(p2, buffer);
```

Comment:
Buffer overflow when actually using the formatting at its length. Don't forgett the Null Terminator.